SystemServices/SystemServices.cpp

```diff
@@ -2228,68 +2228,88 @@ namespace WPEFramework {
 				system("mkdir -p /opt/secure/persistent/System/");
 				LOGWARN(" Territory : Subdirectories created " );
 			}
-			m_strTerritory = "";
-			m_strRegion = "";
+			string regionStr = "";
 			readTerritoryFromFile();//Read existing territory and Region from file
-			ofstream outdata(TERRITORYFILE);
 			string territoryStr = parameters["territory"].String();
 			LOGWARN(" Territory Value : %s ", territoryStr.c_str());
-			if(!outdata){
-				LOGWARN(" Territory : Failed to open the file");
-				returnResponse(resp);
-			}
 			try{
 				if((territoryStr.length() == 3) && (isStrAlphaUpper(territoryStr) == true)){
-					LOGWARN(" Territory : Valid territory name ");
-					outdata << "territory:" + territoryStr+"\n";
-					resp = true;
-				}
-				else {
+					if(parameters.HasLabel("region")){
+						regionStr = parameters["region"].String();
+						if(regionStr != ""){
+							if(isRegionValid(regionStr)){
+								resp = writeTerriytory(territoryStr,regionStr);
+								LOGWARN(" territory name ", territoryStr.c_str());
+								LOGWARN(" region name ", regionStr.c_str());
+							}else{
+								response["message"] = "Invalid region";
+								LOGWARN("Please enter valid region");
+								returnResponse(resp);
+							}
+						}
+					}else{
+						resp = writeTerriytory(territoryStr,regionStr);
+						LOGWARN(" territory name ", territoryStr.c_str());
+					}
+				}else{
+					response["message"] = "Invalid territory";
```

Comment:
Is this within error object. Pls ensure that it's not set within result. It should be under error. For example - {"jsonrpc": "2.0", "result":{"status":false},"error": {"message": "Invalid Territory"}}